module-cluster.tf

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2017, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+# Used to retrieve cluster CA certificate or configure local kube context
+data "oci_containerengine_cluster_kube_config" "oke" {
+  count      = var.create_cluster || coalesce(var.cluster_id, "none") != "none" ? 1 : 0
+  cluster_id = local.cluster_id
+}
+
+locals {
+  cluster_id          = var.create_cluster ? one(module.cluster[*].cluster_id) : var.cluster_id
+  kubeconfig          = try(yamldecode(lookup(one(data.oci_containerengine_cluster_kube_config.oke), "content", "")), { "error" : "yamldecode" })
+  kubeconfig_clusters = try(lookup(local.kubeconfig, "clusters", []), [])
+  kubeconfig_ca_cert  = try(lookup(lookup(local.kubeconfig_clusters[0], "cluster", {}), "certificate-authority-data", ""), "none")
+  cluster_ca_cert     = coalesce(var.cluster_ca_cert, local.kubeconfig_ca_cert)
+}
+
+module "cluster" {
+  count          = var.create_cluster ? 1 : 0
+  source         = "./modules/cluster"
+  compartment_id = local.compartment_id
+  state_id       = random_id.state_id.id
+
+  # Network
+  cni_type                = var.cni_type
+  control_plane_nsg_ids   = setunion(var.control_plane_nsg_ids, var.create_nsgs ? [module.network.cp_nsg_id] : [])
+  control_plane_subnet_id = lookup(module.network.subnet_ids, "cp")
+  control_plane_type      = var.control_plane_type
+  pods_cidr               = var.pods_cidr
```

Comment:
If the CNI is NPN, what should this cidr range be?

Reply:
The value still exists but is ignored with VCN-Native - consistent with the API. Added clarification.